gapis/api/vulkan/frame_loop.go

```diff
@@ -279,6 +301,8 @@ func (f *frameLoop) Transform(ctx context.Context, cmdId api.CmdID, cmd api.Cmd,
 					log.E(ctx, "FrameLoop: Failed to reset changed resources %v.", err)
 					return
 				}
+				// Flush out the reset commands
+				stateBuilder.scratchRes.Free(stateBuilder)
```

Comment:
I see lots of explicit flushing going on here. Can you explain why it is needed?

Reply:
The `scratchRes.Free` will actully write out the mutation: https://github.com/google/gapid/blob/master/gapis/api/vulkan/scratch_resources.go#L55.  This `Free` function is really confusing.